docs/InstrumentationEvents.md

```diff
@@ -25,74 +25,29 @@ Instrumentation Event:
 }
 ```
 
-## <a name="list"> List of available events:
-
-### <a name="consumer"></a> Consumer
-
-* consumer.events.HEARTBEAT  
-  payload: {`groupId`, `memberId`, `groupGenerationId`}
-
-* consumer.events.COMMIT_OFFSETS  
-  payload: {`groupId`, `memberId`, `groupGenerationId`, `topics`}
-
-* consumer.events.GROUP_JOIN  
-  payload: {`groupId`, `memberId`, `leaderId`, `isLeader`, `memberAssignment`, `duration`}
-
-* consumer.events.FETCH_START
-  payload: {}
-
-* consumer.events.FETCH
-  payload: {`numberOfBatches`, `duration`}
-
-* consumer.events.START_BATCH_PROCESS  
-  payload: {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`}
-
-* consumer.events.END_BATCH_PROCESS  
-  payload: {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`, `duration`}
-
-* consumer.events.CONNECT
+Events which have `payload.duration` defined, it represent the time between the current event and the last event.
```

Comment:
That's not quite right, and I don't think there's a single definition you can use for all cases.

* `GROUP_JOIN` - `duration` is the time between starting to join and having joined and synced.
* `FETCH` - Time between `FETCH_START` and receiving _the first_ fetch response.
* `END_BATCH_PROCESS` - Time between `START_BATCH_PROCESS` and this event, meaning the time taken for user-land code to process the batch (plus potential heartbeats and commits).
* `REQUEST` - Time to complete network request (not including potential time waiting to be sent, i.e. `pendingDuration`)

If you want to explain all the properties of the event payloads, I suggest doing it in a nested table or list or something, since even if they share a name they might mean different things.

Reply:
Yes, It was okay with how I ordered them. And saying that duration is the time between last event and this event including everything that happens in between.
But, as we are not going to represent the order, this doesn't make sense. I'll remove it. 👍